gdc_ng_models/models/batch.py

```diff
@@ -0,0 +1,149 @@
+import json
+
+import sqlalchemy
+from sqlalchemy import orm
+from sqlalchemy.ext import declarative
+from sqlalchemy.sql import schema
+
+from gdc_ng_models.models import audit
+
+Base = declarative.declarative_base()
+
+
+class Batch(Base, audit.AuditColumnsMixin):
+    """Batch class to represent a collection of nodes.
+
+    Attributes:
+        id: unique identifier for the batch
+        name: name given to the batch
+        project_id: project the batch is a part of
+        created_datetime: the date and time when the batch is created
+        updated_datetime: the date and time when the batch was last updated
+    """
+
+    __tablename__ = "batch"
+    __mapper_args__ = {"eager_defaults": True}
+    __table_args__ = (
+        schema.PrimaryKeyConstraint("id", name="batch_pk"),
+        schema.Index("batch_name_idx", "name"),
+        schema.Index("batch_project_id_idx", "project_id"),
+        schema.Index("batch_status_idx", "status"),
+    )
+
+    # start sequence at high number to avoid collisions with existing batch_id
+    id_seq = sqlalchemy.Sequence("batch_id_seq", metadata=Base.metadata, start=1000)
+
+    id = sqlalchemy.Column(
+        sqlalchemy.BigInteger,
+        nullable=False,
+        server_default=id_seq.next_value(),
+    )
+    name = sqlalchemy.Column(sqlalchemy.Text, nullable=False)
+    project_id = sqlalchemy.Column(sqlalchemy.Text, nullable=False)
+    status = sqlalchemy.Column(
+        sqlalchemy.Enum("OPEN", "CLOSED", name="batch_status"),
```

Comment:
More recently I started moving away from using Enum because it will require alter statements if we ever need to update the values of the Enum. The need for this is not very obvious when using `create_all`. I prefer using the validate decorator instead.
``` python
@validates("status")
 def validate_status(self, key, status):

        if not status:
            raise ValueError("status is required")

        # ensure all upper cases
        status = status.upper()

        if status not in ["OPEN", "CLOSED"]:
            raise ValueError("invalid status specified")

        return status
```
This way subsequent enum additions will only require code changes